robot/src/robot.py

```diff
@@ -18,9 +19,53 @@ def __init__(self):
 
         self.ultrasonic_sensor = UltrasonicSensor(Port.S4)
 
-    def obstacle_in_front(self):
+        self.routes = {1: [Event("MoveDistance", '{argument: 300}'), Event("Rotate", '{argument: 90}'),
+                           Event("MoveDistance", '{argument: 300}'), Event("Rotate", '{argument: 90}',
+                           Event("MoveDistance", '{argument: 300}'), Event("Rotate", '{argument: 90}'))]}
+
+        self.event_source = event_source
+
+
+    async def set_movement_speed(self, event):
+        speed = event.json.get('argument')
+        self.drive_base.settings(speed=speed)
+
+    # Get the distance traveled by the robot in centimeters
+    async def distance_travelled(self):
+        return self.drive_base.distance() / 10
+
+    async def obstacle_in_front(self):
         return self.ultrasonic_sensor.distance() > 300
 
+    async def stop_moving(self):
+        self.drive_base.stop()
+
+    # Direction == 1 is forward, direction == -1 is backward
+    async def move(self, event):
+        direction = event.json.get('argument')
+        if direction == 1:
+            self.drive_base.drive(200, 0)
+        elif direction == -1:
+            self.drive_base.drive(-200, 0)
+        else:
+            await self.stop_moving()
```

Comment:
```suggestion
    async def move(self, event):
        speed = event.json.get('argument')
        self.drive_base.drive(speed, 0) # We might want to give turn rate as an argument in the json as well.
```

Reply:
Shouldn't we keep direction as well? Otherwise the robot has to rotate 180 degrees to move "backwards"

Reply:
https://pybricks.com/ev3-micropython/robotics.html?highlight=drive%20base#pybricks.robotics.DriveBase.drive
I'm assuming the speed can be negative, thus it should be enough for the backend to send a negative speed for reversing?